kata-agent build failure due to libseccomp

`make build` for IBM Cloud failed with the following errors.

```
  = note: /usr/bin/ld: /root/kata-containers/src/agent/target/x86_64-unknown-linux-musl/release/deps/kata_agent-b16b2ce158e969f5.kata_agent.957fc0a8-cgu.15.rcgu.o: in function `nix::unistd::User::from_uid':
          kata_agent.957fc0a8-cgu.15:(.text._ZN3nix6unistd4User8from_uid17h8596cd70bfdcbe33E+0xb1): warning: Using 'getpwuid_r' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
          /usr/bin/ld: /root/kata-containers/src/agent/target/x86_64-unknown-linux-musl/release/deps/kata_agent-b16b2ce158e969f5.kata_agent.957fc0a8-cgu.15.rcgu.o: in function `tokio::runtime::task::raw::poll':
          kata_agent.957fc0a8-cgu.15:(.text._ZN5tokio7runtime4task3raw4poll17h627f269b34728287E+0x93c): warning: Using 'getaddrinfo' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
          /usr/bin/ld: /tmp/rustck8FPIt/libopenssl_sys-ca4d72d9c99e5536.rlib(b_sock.o): in function `BIO_gethostbyname':
          b_sock.c:(.text.BIO_gethostbyname+0x5): warning: Using 'gethostbyname' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
          /usr/bin/ld: /root/kata-containers/src/agent/target/x86_64-unknown-linux-musl/release/deps/kata_agent-b16b2ce158e969f5.kata_agent.957fc0a8-cgu.15.rcgu.o: undefined reference to symbol 'pthread_mutex_lock@@GLIBC_2.2.5'
          /usr/bin/ld: /lib/x86_64-linux-gnu/libc.so.6: error adding symbols: DSO missing from command line
          collect2: error: ld returned 1 exit status

  = help: some `extern` functions couldn't be found; some native libraries may need to be installed or have their path specified
  = note: use the `-l` flag to specify native libraries to link
  = note: use the `cargo:rustc-link-lib` directive to specify the native libraries to link with Cargo (see https://doc.rust-lang.org/cargo/reference/build-scripts.html#cargorustc-link-libkindname)

error: could not compile `kata-agent` due to previous error
make: *** [Makefile:113: target/x86_64-unknown-linux-musl/release/kata-agent] Error 101
```

This error has been also reported in the upstream kata-agent.
https://github.com/kata-containers/kata-containers/issues/5044

--- a/ibmcloud/image/.gitignore
+++ b/ibmcloud/image/.gitignore
@@ -6,6 +6,7 @@
 /skopeo
 /umoci
 /pause
+/staticlib
 /files/usr/local/bin/agent-protocol-forwarder
 /files/usr/local/bin/kata-agent
 /files/usr/bin/skopeo

--- a/ibmcloud/image/Makefile
+++ b/ibmcloud/image/Makefile
@@ -59,6 +59,12 @@ PAUSE_REPO    = docker://k8s.gcr.io/pause
 PAUSE_VERSION = 3.6
 PAUSE_BUNDLE  = pause_bundle
 
+# Static libseccomp is necessary for kata-agent
+# https://github.com/kata-containers/kata-containers/issues/5044#issuecomment-1239773921
+STATIC_LIB_BUILDER = ../../../kata-containers/ci/install_libseccomp.sh
+STATIC_LIB_DIR     = $(abspath staticlib)
+STATIC_LIB         = $(STATIC_LIB_DIR)/kata-libseccomp/lib/libseccomp.a
+
 build: $(IMAGE_FILE)
 
 push: $(IMAGE_FILE)
@@ -78,9 +84,12 @@ $(AGENT_PROTOCOL_FORWARDER): force
 	cd "$(AGENT_PROTOCOL_FORWARDER_SRC)" && $(MAKE) agent-protocol-forwarder
 	install -D --compare "$(AGENT_PROTOCOL_FORWARDER_SRC)/agent-protocol-forwarder" "$@"
 
-$(KATA_AGENT): force
-	cd "$(KATA_AGENT_SRC)" && $(MAKE) BUILD_TYPE=$(KATA_AGENT_BUILD_TYPE)
+$(KATA_AGENT): force $(STATIC_LIB)
+	cd "$(KATA_AGENT_SRC)" && $(MAKE) BUILD_TYPE=$(KATA_AGENT_BUILD_TYPE) LIBSECCOMP_LINK_TYPE=static LIBSECCOMP_LIB_PATH=$(dir $(STATIC_LIB))
 	install -D --compare "$(KATA_AGENT_SRC)/target/$(shell uname -m)-unknown-linux-$(if $(findstring s390x,$(shell uname -m)),gnu,musl)/$(KATA_AGENT_BUILD_TYPE)/$(@F)" "$@"
+
+$(STATIC_LIB):
+	$(STATIC_LIB_BUILDER) $(STATIC_LIB_DIR)/kata-libseccomp $(STATIC_LIB_DIR)/kata-gperf
 
 $(SKOPEO_SRC):
 	git clone -b "v$(SKOPEO_VERSION)" "$(SKOPEO_REPO)" "$(SKOPEO_SRC)"

